src/hooks/data/use-upsert.ts

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+
+import { Count, Filter, PostgrestError, Returning } from '../../types'
+import { useClient } from '../use-client'
+import { initialState } from './state'
+
+export type UseUpsertState<Data = any> = {
+    count?: number | null
+    data?: Data | Data[] | null
+    error?: PostgrestError | null
+    fetching: boolean
+}
+
+export type UseUpsertResponse<Data = any> = [
+    UseUpsertState<Data>,
+    (
+        values: Partial<Data>,
```

Comment:
Let's have this match `execute`'s signature.

```suggestion
        values: Partial<Data> | Partial<Data>[],
```

Reply:
Awesome finding :raised_hands: I haven't noticed that one